lib/evmone/analysis.cpp

```diff
@@ -124,7 +135,41 @@ code_analysis analyze(
             instr.arg.p.number = static_cast<int>(i);
         else if (c >= OP_LOG0 && c <= OP_LOG4)
             instr.arg.p.number = c - OP_LOG0;
-        else if (is_terminator(c))
+
+        if (rev >= EVMC_ISTANBUL)
```

Comment:
I think this should be better `else if`, so that it's not even considered if it's some other opcode.